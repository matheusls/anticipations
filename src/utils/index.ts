export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};
