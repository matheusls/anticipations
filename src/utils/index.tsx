import { toast, TypeOptions } from 'react-toastify';

export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};

type ErrorToastProps = {
  text: string;
  type: TypeOptions;
};

export const renderErrorToast = (error = '') => {
  const errorToastProps: ErrorToastProps = {
    '500': {
      text: 'Oops, tivemos um problema. Por favor, tente novamente.',
      type: 'error' as const,
    },
    '408': {
      text: 'Oops, nossos servidores estão demorando para responder...',
      type: 'warning' as const,
    },
    offline: {
      text: 'Você parece estar com problemas de conexão!',
      type: 'warning' as const,
    },
  }[error] || {
    text: 'Oops, houve um problema. Por favor, tente novamente.',
    type: 'error' as const,
  };

  const { text, type } = errorToastProps;

  return toast(text, {
    position: 'top-center',
    autoClose: false,
    closeOnClick: true,
    type,
  });
};
