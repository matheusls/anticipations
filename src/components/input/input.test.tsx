import { fireEvent, render, screen } from 'tests/utils';

import Input from './input';

const inputProps = {
  id: 'some-input',
  name: 'some-input',
  placeholder: 'Input Component Placeholder',
  type: 'text',
};

describe('<Input />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Input {...inputProps} />);

    expect(container).toMatchSnapshot();
  });

  it('should render change Input value correctly', () => {
    render(<Input {...inputProps} />);

    const inputValue = 'Input Component Value';
    const input = screen.getByPlaceholderText(/input component placeholder/i);

    fireEvent.change(input, { target: { value: inputValue } });

    expect(input).toHaveValue(inputValue);
  });

  it('should render change Input border color on focus', async () => {
    render(<Input {...inputProps} />);

    const input = screen.getByPlaceholderText(/input component placeholder/i);

    input.focus();

    expect(input).toHaveFocus();
  });
});
