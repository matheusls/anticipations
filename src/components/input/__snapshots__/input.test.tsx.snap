// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
.c0 {
  color: #000;
  border: 1px solid;
  border-color: #dde6e9;
  border-radius: 4px;
  display: block;
  font-size: 0.875rem;
  line-height: 1;
  margin-bottom: 3px;
  padding: 10px 14px;
  width: 100%;
}

.c0:focus {
  border-color: #66afe9;
  outline: none;
}

<div>
  <input
    class="c0"
    id="some-input"
    name="some-input"
    placeholder="Input Component Placeholder"
    type="text"
  />
</div>
`;
